sklearn/utils/estimator_checks.py

```diff
@@ -1586,33 +1587,22 @@ def check_estimators_fit_returns_self(name, estimator_orig,
 def check_estimators_unfitted(name, estimator_orig):
     """Check that predict raises an exception in an unfitted estimator.
 
-    Unfitted estimators should raise either AttributeError or ValueError.
-    The specific exception type NotFittedError inherits from both and can
-    therefore be adequately raised for that purpose.
+    Unfitted estimators should raise a NotFittedError.
     """
 
     # Common test for Regressors, Classifiers and Outlier detection estimators
     X, y = _boston_subset()
 
     estimator = clone(estimator_orig)
 
-    msg = "fit"
+    msg = ("{} instance is not fitted yet. Call 'fit' with appropriate "
```

Comment:
Making this requirement more strict also places the same constraint on developers who use check_estimator. For other kinds of backwards compatibility issues we would raise a deprecation warning and still pass the test, rather than raise an error outright. I would consider just strengthening this test if we don't think developers will be upset about having their compatibility broken all of a sudden. In any case, this change needs to be separately documented under "Changes to estimator checks"

Reply:
On the one hand, I think that you are completely right and it would not cost that much to revert to the previous match (@agamemnonc could you revert it).

On the other hand, it happens in the past that scikit-learn broke some of the backward compatibility in the common test and I personally did not mind because it was always toward more consistent checks. For instance, if the estimator tags come to break some of the estimator checks, I would think this for the best since it would allow more modularity.

Reply:
Please could you clarify whether you would like me to revert to previous state or include the new error message with a deprecation warning for the old one.

Also some leads re: the rst file "Changes to estimator checks" update would be appreciated, thanks.